.scripts/process_build.sh

```diff
@@ -0,0 +1,175 @@
+  unzip -o $MAJOR_ZIP_FILE
+else
+  TYPE="_Linux_x86_64.tar.gz"
+  MAJOR_TAR_FILE=fstar_$CURRENT_VERSION$TYPE
+  if [[ -f $MAJOR_TAR_FILE ]]; then
+    MINOR_TAR_FILE=fstar_$TIME_STAMP$TYPE
+    cp $MAJOR_TAR_FILE $MINOR_TAR_FILE
+    tar -x -f $MAJOR_TAR_FILE
+  else
+    echo -e "* ${RED}FAIL!${NC} src/ocaml-output/make package did not create "$MAJOR_ZIP_FILE" or "$MAJOR_TAR_FILE
+    exit 1
+  fi
+fi
+
+echo "*** Make the examples ***"
+cd fstar
+make -C examples/micro-benchmarks 1> MicroBenchMarkOutput.log
+make -C examples/hello ocaml 1> HelloOcamlOutput.log
+make -C examples/hello fs 1> HelloFStarOutput.log
+make -j6 -C examples 1> AllExamples.log
+
+echo "*** Verify the examples ***"
+echo "-- Verify Micro-benchmarks -- should output 'Success:' *"
+if ! egrep 'Success:' MicroBenchMarkOutput.log; then
+  echo -e "* ${RED}FAIL!${NC} for examples/micro-benchmarks - Success: was not found in MicroBenchMarkOutput.log"
+  exit 1
+else
+  echo -e "* ${GREEN}PASSED!${NC} for examples/micro-benchmarks"
+fi
+
+echo "-- Verify hello ocaml -- should output Hello F*! *"
+if ! egrep 'F*!' HelloFStarOutput.log; then
+  echo -e "* ${RED}FAIL!${NC} for examples/hello ocaml - F*! was not found in HelloOcamlOutput.log"
+  exit 1
+else
+  echo -e "* ${GREEN}PASSED!${NC} for examples/hello ocaml"
+fi
+
+echo "-- Verify hello fs -- should output Hello F*!"
+if ! egrep 'F*!' HelloFStarOutput.log; then
+  echo -e "* ${RED}FAIL!${NC} for examples/hello fs - F*! was not found in HelloFStarOutput.log"
+  exit 1
+else
+  echo -e "* ${GREEN}PASSED!${NC} for examples/hello fs"
+fi
+
+echo "-- Verify all examples -- Look for Success:"
+if ! egrep 'Success:' AllExamples.log; then
+  echo -e "* ${RED}FAIL!${NC} for all examples - Success: was not found in AllExamples.log"
+  exit 1
+else
+  echo -e "* ${GREEN}PASSED!${NC} for all examples"
+fi
+
+# Got to this point, so know it passed - copy minor version out
+echo "*** Upload the minor version of the package. Will only keep the most recent 4 packages ***"
+cd $ORIG_PWD
+BN_BINARYSPATH_ROOT=~/binaries
+BN_BINARYSPATH=$BN_BINARYSPATH_ROOT/weekly
+BN_FILESTOKEEP=4
+
+if [[ ! -d $BN_BINARYSPATH_ROOT ]]; then
+  cd ~
+  git clone https://github.com/FStarLang/binaries.git
+fi
+
+cd $BN_BINARYSPATH_ROOT
+git checkout master
+git pull origin master
+
+echo "-- copy files and add to Github --"
+if [[ -f $ORIG_PWD/src/ocaml-output/$MINOR_ZIP_FILE ]]; then
+  echo "-- "$ORIG_PWD/src/ocaml-output/$MINOR_ZIP_FILE $BN_BINARYSPATH
+  cp $ORIG_PWD/src/ocaml-output/$MINOR_ZIP_FILE $BN_BINARYSPATH
+  cd $BN_BINARYSPATH
+  git add $MINOR_ZIP_FILE
+  cd ..
+fi
+if [[ -f $ORIG_PWD/src/ocaml-output/$MINOR_TAR_FILE ]]; then
+  echo "--" $ORIG_PWD/src/ocaml-output/$MINOR_TAR_FILE $BN_BINARYSPATH
+  cp $ORIG_PWD/src/ocaml-output/$MINOR_TAR_FILE $BN_BINARYSPATH
+  git add $BN_BINARYSPATH/$MINOR_TAR_FILE
+fi
+
+# Now that latest package is added, remove the oldest one because only keeping most recent 4 packages
```

Comment:
These two pieces of code could also go into a function to avoid code duplication... something like (sketch):

```
function cleanup_files () {
  local suffix=$1
  local files=$BN_BINARYSPATH/*.$suffix
  local file_count=$(ls -1 $files 2>/dev/null | wc -l)
  if (( $file_count > $BN_FILESTOKEEP )); then
    # Windows git rm just needs the file name and fails if give path so just get file name
    local file_list=$(ls -t1 $files | xargs -n1 basename | tail -n +$(($BN_FILESTOKEEP+1)))
    for f in $file_list; do
      git rm $f
    done
  fi
}
```

This sketch tries to address the various nits below: